controllers/cloudinit/scripts/00-install-microk8s.sh

```diff
@@ -1,12 +1,20 @@
 #!/bin/bash -xe
 
 # Usage:
-#   $0 $microk8s_snap_channel
+#   $0 $microk8s_snap_channel $confinement
 #
 # Assumptions:
 #   - snap is installed
 
-while ! snap install microk8s --classic --channel "${1}"; do
+install_cmd="snap install microk8s --channel"
+
+if [[ "${2}" == "strict" ]]; then
+  install_cmd="$install_cmd "${1}"-strict"
+else
+  install_cmd="$install_cmd "${1}" --classic"
+fi
```

Comment:
I'd rather we move this logic into the Go code that generates the template instead of the bash script.

Reply:
Did this. Have a look.